### PR TITLE
#96 Updated terraform for openApi

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #96 
- Added terraform and scripts for preparing and importing openApi to ApiGateway.
- Removed api_gw module from Terraform. cause the API gateway integration will be defined in openApi.
- Updated docker-compose file for merged openApi file to run swagger locally
- also, for now, added an example code for auth/register-org path (which has a separate ticket #75)

for openApi files please look at the previous PR #97 

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #97

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [ ] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,9 +9,15 @@ RUN apt update && apt install -y ca-certificates curl gnupg make gcc zip unzip a
     python3-sphinx graphviz \
     --no-install-recommends
 
+# AWS CLI v2
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "/tmp/awscliv2.zip" && \
```
Comment (melon-ruet) [inline Dockerfile:13]: There is a pip version of awscliv2
https://pypi.org/project/awscliv2/

File: Dockerfile [hunk 2]
```diff
@@ -36,4 +42,4 @@ RUN curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /
 RUN apt update && apt install -y nodejs --no-install-recommends
 
 # Install spectral: API linter
-RUN npm install -g @stoplight/spectral-cli @redocly/cli@latest
+RUN npm install -g @stoplight/spectral-cli @redocly/cli@latest swagger-cli
```
Comment (melon-ruet) [inline Dockerfile:45]: If we use swagger-cli, we do not need @redocly/cli@latest

File: iac/terraform/aws/auth/lambdas.tf [hunk 3]
```diff
@@ -8,6 +8,15 @@ locals {
...
         foo = "bar"
       }
+    },
+    register-org = {
+      name     = "register-org"
+      zip_path = "${path.module}/../../../../core/services/aws/.build/zips/registerOrg.zip"
+      policy   = data.aws_iam_policy_document.lambda_common_policy
+      handler  = "registerOrg.default"
+      env_variables = {
+        foo = "bar"
```
Comment (melon-ruet) [inline iac/terraform/aws/auth/lambdas.tf:18]: Why do we need env var placeholder?

File: openapi/Dockerfile
Comment (melon-ruet) [inline openapi/Dockerfile:1]: Why do we need this file?

File: openapi/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+version: "3.8"
+
+services:
+  swagger-cli:
```
Comment (melon-ruet) [inline openapi/docker-compose.yml:4]: Why is this necessary?

File: openapi/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      - SWAGGER_JSON=/openapi/mergedOpenApi.json
+    ports:
+      - "8083:8080"
+    volumes:
+      - ./:/openapi
+    depends_on:
+      - swagger-cli
+    restart: always
+
+  redoc:
```
Comment (melon-ruet) [inline openapi/docker-compose.yml:27]: This is unnecessary

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: openapi/Dockerfile
Comment (mahinuzzaman) [reply openapi/Dockerfile:1]: this file is for running a container that will create the combined openapi file

File: openapi/docker-compose.yml (hunk 4 above)
Comment (mahinuzzaman) [reply openapi/docker-compose.yml:4]: this part is for running a container that will create the combined openapi file

File: iac/terraform/aws/auth/lambdas.tf (hunk 3 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/auth/lambdas.tf]: this is list of environment variables for lambda. this was example variable. removed it

File: openapi/docker-compose.yml (hunk 5 above)
Comment (mahinuzzaman) [reply openapi/docker-compose.yml]: removed it. updated in 7b25eb673cca15ddae3e924df99684249951f607

**Comment by mahinuzzaman** [comment]
@melon-ruet removed docker-compose file in 0d58d235716eecc4d2c243c6caf6b1fdafb199c9. will update openapi local process in a separate pr
